src/nft/utils/ipfs.ts

```diff
@@ -0,0 +1,31 @@
+export function IsValidIpfsUrl(url: string): boolean {
+  try {
+    const parsedUrl = new URL(url);
+    const isValidIpfsUrl = parsedUrl.protocol === 'ipfs:';
+    return isValidIpfsUrl;
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Converts an IPFS URL to a standardized HTTPS URL using a public IPFS gateway.
+ *
+ * @param url The original URL
+ * @returns https:// URL for the media
+ * @link https://docs.ipfs.tech/how-to/address-ipfs-on-web/ - for more details.
+ */
+export function convertIpfsToHttps(url?: string): string | undefined {
+  if (!url) {
+    return undefined;
+  }
+
+  const IPFS_GATEWAY_URL =
+    'https://b891d14d436694bb9a7feeba91730b95.ipfscdn.io/ipfs/';
```

Comment:
is there rate limiting on this gateway?

Reply:
i'm not sure, this is the one wallet uses

Reply:
the only time I've seen that we've had to handle ipfs urls on the FE is for cb pre-mints, shouldn't happen often

Reply:
could switch to just ipfs.io